### PR TITLE
Mv level dirs

### PR DESCRIPTION
**Pull request by matthewvon**
Code to separate level table files into directories by "level".

### CODE REVIEW

File: db/repair.cc [hunk 1]
```diff
@@ -369,21 +439,33 @@ class Repairer {
...
-  void ArchiveFile(const std::string& fname) {
+  void ArchiveFile(const std::string& fname, bool two_levels=false) {
     // Move into another directory.  E.g., for
     //    dir/foo
     // rename to
     //    dir/lost/foo
-    const char* slash = strrchr(fname.c_str(), '/');
+    std::string::size_type slash, slash2;
+
+    slash=fname.rfind('/');
```
Comment (gburd) [inline db/repair.cc:449]: rfind always scares me a bit, is there a better way?  What's going on here?

File: tools/repair.cc [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    return( error_seen && 0!=error_counter ? 1 : 0 );
+
+}   // main
+
+
+void
+command_help()
+{
+    fprintf(stderr, "bloom_scan [option | data_base]*\n");
```
Comment (gburd) [inline tools/repair.cc]: I the name of this tool "bloom_scan"?

**Comment by gburd** [comment]
Everything but the conversion routine was reviewed successfully.  MvM is going to add some more logic to manage the repair process in cases where the migration process fails mid-way.  Once that's reviewed this is ready to go.

**Comment by gburd** [comment]
+1 to merge

**Comment by matthewvon** [comment]
Thank you.

You in the mood to do one more.  Andy does not seem to be getting around to the mv-perf-counters.  I can do a call on it pretty much anytime.  This is my last contribution to 1.3.

Matthew

**Pull request merged**